### PR TITLE
Bugfix/restore contributions

### PR DESCRIPTION
**Pull request by chime-mu**
@jlopker: This is the fix for the deleted contributions and cancelled buckets. Can you review?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ before_script:
   - cp config/database.yml.travis config/database.yml
   - bundle exec rake db:setup
 addons:
+  postgresql: "9.5"
```
Comment (jlopker) [inline .travis.yml]: without this, the rake task wouldn't be able to compute sql commands?

File: app/controllers/buckets_controller.rb [hunk 2]
```diff
@@ -26,7 +26,7 @@ def create
   api :PATCH, '/buckets/:id', 'Update a bucket'
   def update
     bucket = Bucket.with_totals.find(params[:id])
-    render status: 403, nothing: true and return unless bucket.is_editable_by?(current_user) && !bucket.archived?
+    render status: 403, nothing: true and return unless bucket.is_editable_by?(current_user) && (bucket.is_idea? || bucket.is_funding?)
```
Comment (jlopker) [inline app/controllers/buckets_controller.rb]: buckets can be updated when they are funded as well

File: lib/tasks/greaterthan.rake [hunk 3]
```diff
@@ -80,4 +80,81 @@ namespace :greaterthan do
...
+      end
+
+      # Add contributions
+      sql = "INSERT INTO contributions(id, user_id, bucket_id, created_at, updated_at, amount) VALUES \n" <<
+        missining_contributions.map {|l| %((#{l[:id]}, #{l[:user_id]}, #{l[:bucket_id]}, '#{l[:created_at]}', '#{l[:updated_at]}', #{l[:amount]}),)}.
+          reduce("",:<<)
+      sql[-1] = ';'
+      ActiveRecord::Base.connection.execute(sql)
+
+      # Remove the fact the buckets was deleted
```
Comment (jlopker) [inline lib/tasks/greaterthan.rake]: # Remove the archived data from these buckets

**Comment by jlopker** [comment]
@chime-mu wrote a few comments but otherwise it looks good!

File: .travis.yml (hunk 1 above)
Comment (chime-mu) [reply .travis.yml]: @jlopker Yes. I've added a jsonb datatype which doesn't exist in Postgresql 9.1

**Pull request merged**